docs/Using Fleet/MDM-macOS-setup.md

```diff
@@ -339,6 +343,89 @@ Testing requires a test Mac that is present in your Apple Business Manager (ABM)
 
 4. Boot up your test Mac and complete the custom out-of-the-box setup experience.
 
+## Populate the Full Name and Account Name
+
+When an end user unboxes their Mac, they're presented with the **Create a Computer Account** pane where they choose their local Full Name, Account Name, and Password they'll used to login to your Mac.
+
+With Fleet, you can populate the Full Name and Account Name with information from your Identity Provider (IdP).
+
+To populate the Full Name and Account Name, we will do the following steps:
+
+1. Enable end user authentication
+2. Configure your IdP
+3. Edit your macOS Setup Assistant settings
+4. Release the Mac from Await Configuration
+
+### Step 1: enable end user authentication
+
+If you haven't already, follow the steps to enable end user authentication [here](#end-user-authentication-and-eula).
+
+### Step 2: configure your IdP
+
+1. In your IdP, make sure the end user's full name is set to one of the following attributes (depends on IdP): `name`, `displayname`, `cn`, `urn:oid:2.5.4.3`, or `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name`. Fleet will populate the Account Name with any of these.
+
+2. In your IdP configuration set **Name ID** to the end user's email. Fleet will trim this email and use it to populate the Account Name. For example, a "johndoe@example.com" email turn into a "johndoe" Account Name.
+
+### Step 3: edit your macOS Setup Assistant settings
+
+In Fleet, you control your macOS Setup Assistant settings with an automatic enrollment profile.
+
+Set `await_device_configured` to `true` in your automatic enrollment profile. Learn how to edit your automatic enrollment profile [here](#macos-setup-assistant). 
+
+This will tell the Mac to not allow the user through Setup Assistant until the Mac is released. This way, the Mac has time to get the Full Name and Account Name from your IdP before the **Create a Computer Account** pane.
+
+## Step 4: release the Mac from Await Configuration
+
+Releasing the Mac can be automated using the Fleet API. In this example, we'll release a test Mac manually.
```

Comment:
Is this the right way to present this step? I don't get a good sense of how I would automate this because it's so clicky. To automate this, I need to set up some server that accepts a signal from Fleet or the host that it's ready to go. Then I need to trigger the command. How do I do that?

Reply:
Same with the steps below.

Reply:
> I don't get a good sense of how I would automate this because it's so clicky.

Great point. I updated the instructions to address this in my latest commits. 

> I need to set up some server that accepts a signal from Fleet or the host that it's ready to go. 

I don't describe setting up a server and how. I think this can be addressed in a later iteration.

What do you think?

Reply:
Hi Noah, yeah I don't think we need to tell users how to set up a server, but we should tell them what the interface is between our app and their server.

Reply:
> what the interface is between our app and their server.

For example, the Fleet API?